src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java

```diff
@@ -179,13 +183,8 @@ private void hideDetails(@RequestBody TextSubmission textSubmission) {
                                                                       @RequestParam(defaultValue = "false") boolean submittedOnly) {
         log.debug("REST request to get all TextSubmissions");
         Exercise exercise = exerciseService.findOneLoadParticipations(exerciseId);
-        Course course = exercise.getCourse();
-        User user = userService.getUserWithGroupsAndAuthorities();
-        if (!authCheckService.isTeachingAssistantInCourse(course, user) &&
-            !authCheckService.isInstructorInCourse(course, user) &&
-            !authCheckService.isAdmin()) {
-            return forbidden();
-        }
+
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(exercise)) return forbidden();
```

Comment:
Please also review the code in this method and try to avoid extensive use of stream, peek, map and filter with hidden database calls. Rework the code in the style of how other code is written and try to put everything into one or two database calls.

Reply:
Also the JavaDoc is not very intuitive. By the way: I think you might be able to merge this request with the two below by introducing path variables

Reply:
I have added a TODO and assigned it to me, since this method was already existing. I will address the refactor in another PR